bc-level LargeArray for NUMA-aware node data; burn power; prealloc formula adjustment

l-hoang opened the pull request:
BC node data now stored in a LargeArray for some NUMA awareness with
node data access.

arthurp commented on python/galois/connected_components_property_graph.py:
```diff
@@ -0,0 +1,157 @@
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--algoType', type=str, default="push")
+    parser.add_argument('--propertyID', type=int, default=0)
```
Remove propertyID. It's never used.

gurbinder533 commented on python/galois/connected_components_property_graph.py:
```diff
@@ -0,0 +1,157 @@
+    for ii in graph.edges(nid):
+        dst = graph.get_edge_dst(ii)
+        # Pull the minimum component from your neighbors
+        if(comp_current[nid] > comp_current[dst]):
```
I try hard but they just slip in once in a while :P.

l-hoang commented:
If this commit is to be kept, need to remove the creation of node data from the typed property graph.
For now it's not done.

l-hoang commented:
NUMA helps on Intel machine, so keeping it + cleaned up PropertyGraph node type. Also changed
default chunk size to 64 (seems to help; if it doesn't, I'll change it back).

Obviously this is a very input dependent setting though....

l-hoang commented:
Added some wizardry (burnPower) to the app: @nicelhc13 please check performance.

Commit:
BC: LargeArray data, burn power CLI, prealloc

BC node data now stored in a LargeArray for some NUMA awareness with
node data access. Allows turning on burnPower option for threads
(good for small parallel sections, e.g., road networks) for Level
via command line option. Slight adjustment of prealloc formula
to better suit the road-network (may need better readjustment
in the future for larger graphs).

l-hoang commented:
CLI option for burnPower, slight adjustment of prealloc formula as well.

roshandathathri commented on lonestar/analytics/cpu/betweennesscentrality/betweenness_centrality_cli.cpp:
```diff
@@ -68,6 +68,12 @@ static cll::opt<BetweennessCentralityPlan::Algorithm> algo(
+static cll::opt<bool> thread_spin(
+    "threadSpin",
+    cll::desc("If enabled, threads busy-wait for work rather than use "
+              "condition variable (default false)"),
```
This seems like a library level option rather than an app specific option

l-hoang replied:
that's fair; might be better to move it to a common CLI in the lonestar init later

roshandathathri reviewed (approved):
LGTM
